tools/rewriter/GenCallAsm.cpp

```diff
@@ -1160,10 +1178,27 @@ std::string emit_asm_wrapper(AbiSignature sig,
 
   emit_set_return_pkru(aw, caller_pkey, arch);
 
-  // Call the post-condition function, if one was specified.
-  // The call happens in the caller's compartment.
-  if (target_post_condition_name) {
-    emit_post_condition_fn_call(aw, arch, *target_post_condition_name);
+  // Call the post-condition function for this target function.
+  // The calls happens in the caller's compartment.
+  if (!post_conditions.empty()) {
+    // Args are callee saved, so we need to restore them.
+    emit_restore_args(aw, arch);
+
+    add_comment_line(aw, "Align stack");
+    if (arch == Arch::X86) {
+      add_asm_line(aw, "subq $8, %rsp");
```

Comment:
Doesn't this and the `addq $8, %rsp` below depend on the number of arguments?

Reply:
I agree, it isn't clear to me what the alignment of the stack is going to be when we hit this code. That said, this PR isn't changing the effective control flow here from what I can tell--it's just inlining this from `emit_post_condition_fn_call`.

Reply:
From offline discussions, it also seems like this shouldn't depend on the number of arguments, since we always assume 6 args.  There are still some issues with stack alignment, though, but I'll fix those in a separate PR instead of at the base of this PR stack.